axonserver/src/main/java/io/axoniq/axonserver/transport/grpc/EventProcessorGrpcController.java

```diff
@@ -112,8 +125,14 @@ public void startEventProcessor(EventProcessorIdentifier eventProcessorId,
     @Override
     public void splitEventProcessor(EventProcessorIdentifier processorId,
                                     StreamObserver<AdminActionResult> responseObserver) {
-        service.split(new EventProcessorIdMessage(contextProvider.getContext(), processorId),
-                      new GrpcAuthentication(authenticationProvider))
+        EventProcessorIdMessage eventProcessorIdMessage;
+        if (Strings.isNotBlank(processorId.getContextName())) {
+            eventProcessorIdMessage = new EventProcessorIdMessage(processorId.getContextName(), processorId);
```

Comment:
You can move this logic to the EventProcessorIdMessage constuctor, so you don't have to repeat it for every request.